internal/git/git.go

```diff
@@ -339,8 +339,75 @@ func (g *GitOperations) runCmd(ctx context.Context, directory string, args ...st
 
 	if err := cmd.Wait(); err != nil {
 		// exitErr := err.(*exec.ExitError)
-		return "", stderrBuf.String(), err
+		return stdoutBuf.String(), stderrBuf.String(), err
 	}
 
 	return stdoutBuf.String(), stderrBuf.String(), nil
 }
+
+// replace lines 348 to 386
+func (g *GitOperations) HasConflict(ctx context.Context, proposedBranch, activeBranch string) (bool, error) {
+	logger := log.FromContext(ctx)
+	repoPath := gitpaths.Get(g.gap.GetGitHttpsRepoUrl(*g.gitRepo) + g.pathContext)
+
+	// Fetch both branches from origin
+	if _, stderr, err := g.runCmd(ctx, repoPath, "fetch", "origin", activeBranch, proposedBranch); err != nil {
+		logger.Error(err, "could not fetch branches", "stderr", stderr)
+		return false, err
+	}
+
+	// Checkout the active branch
+	if _, stderr, err := g.runCmd(ctx, repoPath, "checkout", "--progress", "-B", activeBranch, "origin/"+activeBranch); err != nil {
+		logger.Error(err, "could not checkout active branch", "branch", activeBranch, "stderr", stderr)
+		return false, fmt.Errorf("failed to checkout active branch %q: %w", activeBranch, err)
+	}
+
+	// Merge the proposed branch without committing
+	stdout, stderr, err := g.runCmd(ctx, repoPath, "merge", "--no-commit", "--no-ff", "origin/"+proposedBranch)
+	conflictDetected := strings.Contains(stdout, "CONFLICT")
+
+	// Always attempt to abort the merge to clean up
+	if _, abortStderr, abortErr := g.runCmd(ctx, repoPath, "merge", "--abort"); abortErr != nil {
+		if !strings.Contains(abortStderr, "MERGE_HEAD missing") { // Ignore the error if there is no merge in progress
+			logger.Error(abortErr, "could not abort merge", "stderr", abortStderr)
+			return false, fmt.Errorf("failed to abort merge: %w", abortErr)
+		}
+	}
+
+	if err != nil && conflictDetected {
```

Comment:
Is the err check necessary? Could we just check conflictDetected and then return early?